File: Extension/src/app/domain/bookmarks/entities/bookmark.ts
export class Bookmark
{
	public DateTimeAddedUTC: string;

	public Description: string;

	/**
	 * A UUID that's generated for each bookmark.
	 */
	public Id: string;

	/**
	 * The index of the bookmark, used for sorting.
	 */
	public Index: number;

	public Note: string;

	/**
	 * The parent collection Id of the bookmark.
	 */
	public ParentId: string;

	public Title: string;

	public Url: string;
}

File: API/Bookmarx.Shared/v1/Bookmarks/Entities/Bookmark.cs
using Google.Cloud.Firestore;

namespace Bookmarx.Shared.v1.Bookmarks.Entities;

[FirestoreData]
public class Bookmark
{
	public Bookmark()
	{
		// Needed for firebase
	}

	[FirestoreProperty]
	public DateTime DateTimeAddedUTC { get; set; }

	[FirestoreProperty]
	public string Description { get; set; } = string.Empty;

	[FirestoreProperty]
	public string Id { get; set; }

	[FirestoreProperty]
	public int Index { get; set; }

	[FirestoreProperty]
	public string Note { get; set; } = string.Empty;

	[FirestoreProperty]
	public string ParentId { get; set; }

	[FirestoreProperty]
	public string Title { get; set; }

	[FirestoreProperty]
	public string Url { get; set; }
}